crates/compiler/gen_dev/src/generic64/mod.rs

```diff
@@ -1651,6 +1652,82 @@ impl<
         self.free_symbol(&Symbol::DEV_TMP5);
     }
 
+    fn build_list_concat(
+        &mut self,
+        dst: &Symbol,
+        args: &'a [Symbol],
+        arg_layouts: &[InLayout<'a>],
+        element_layout: InLayout<'a>,
+        ret_layout: &InLayout<'a>,
+    ) {
+        let list_a = args[0];
+        let list_a_layout = arg_layouts[0];
+        let list_b = args[1];
+        let list_b_layout = arg_layouts[1];
+
+        // Load list alignment argument (u32).
+        let u32_layout = Layout::U32;
+        let list_alignment = self.layout_interner.alignment_bytes(*ret_layout);
+        self.load_literal(
+            &Symbol::DEV_TMP,
+            &u32_layout,
+            &Literal::Int((list_alignment as i128).to_ne_bytes()),
+        );
+
+        // Load element_width argument (usize).
+        let u64_layout = Layout::U64;
+        let element_width = self.layout_interner.stack_size(element_layout);
+        self.load_literal(
+            &Symbol::DEV_TMP2,
+            &u64_layout,
+            &Literal::Int((element_width as i128).to_ne_bytes()),
+        );
+
+        // Setup the return location.
+        let base_offset = self
+            .storage_manager
+            .claim_stack_area(dst, self.layout_interner.stack_size(*ret_layout));
+
+        let lowlevel_args = bumpalo::vec![
+        in self.env.arena;
+            list_a,
+            list_b,
+            // alignment
+            Symbol::DEV_TMP,
+            // element_width
+            Symbol::DEV_TMP2,
+         ];
+        let lowlevel_arg_layouts = bumpalo::vec![
+        in self.env.arena;
+            list_a_layout,
+            list_b_layout,
+            u32_layout,
+            u64_layout
+        ];
```

Comment:
style thing: I believe you can just drop the `bumpalo::vec!` and the `in self.env.arena;` bits; then what you get is an array, but `build_fn_call` accepts slices so that should work out.